secure_enclave/TEUtils.cpp

```diff
@@ -65,152 +70,150 @@ std::string fieldElementToString(const T &field_elem, int base = 10) {
+ * Converts inidivudally each field element to string, and concatenates them.
+ * Total size will always be 256 (64 * 4)
+ */
+std::string G2ToString(libff::alt_bn128_G2 elem) {
+  std::string pkey_str;
 
-  std::string result = "";
+  elem.to_affine_coordinates();
 
-  try {
+  pkey_str += fieldElementToHex(elem.X.c0);
+  pkey_str += fieldElementToHex(elem.X.c1);
+  pkey_str += fieldElementToHex(elem.Y.c0);
+  pkey_str += fieldElementToHex(elem.Y.c1);
 
-    result += fieldElementToString(elem.X.c0);
-    result += delim;
-    result += fieldElementToString(elem.X.c1);
-    result += delim;
-    result += fieldElementToString(elem.Y.c0);
-    result += delim;
-    result += fieldElementToString(elem.Y.c1);
+  return pkey_str;
+}
+
+std::string convertHexToDec(char *hex_str) {
+  mpz_t dec;
+  mpz_init(dec);
 
-    return result;
+  std::string output;
 
+  try {
+    std::string hex(hex_str, 64);
+    if (mpz_set_str(dec, hex.c_str(), 16) == -1) {
+      mpz_clear(dec);
+      LOG_ERROR("Bad formatted hex string provided");
+      return output;
+    }
+
+    char arr[mpz_sizeinbase(dec, 10) + 2];
+    char *tmp = mpz_get_str(arr, 10, dec);
+
+    output = tmp;
   } catch (std::exception &e) {
     LOG_ERROR(e.what());
-    return result;
   } catch (...) {
-    LOG_ERROR("Unknown throwable");
-    return result;
+    LOG_ERROR("Exception in convert hex to dec");
   }
 
-  return result;
+  mpz_clear(dec);
+  return output;
 }
 
-std::vector<libff::alt_bn128_Fq> SplitStringToFq(const char *coords,
-                                                 const char symbol) {
-  std::vector<libff::alt_bn128_Fq> result;
-  std::string str(coords);
-  std::string delim;
+libff::alt_bn128_G2 stringToG2(char *str, size_t size) {
+  if (size != CIPHERTEXT_CHARACTER_LENGTH) {
+    LOG_ERROR("Wrong string size to convert to G2");
+  }
 
-  CHECK_ARG_CLEAN(coords);
+  libff::alt_bn128_G2 ret;
 
-  try {
+  ret.Z = libff::alt_bn128_Fq2::one();
+
+  ret.X.c0 = libff::alt_bn128_Fq(convertHexToDec(str).c_str());
+  ret.X.c1 = libff::alt_bn128_Fq(convertHexToDec(str + 64).c_str());
+  ret.Y.c0 = libff::alt_bn128_Fq(convertHexToDec(str + 128).c_str());
+  ret.Y.c1 = libff::alt_bn128_Fq(convertHexToDec(str + 192).c_str());
+
+  return ret;
+}
 
-    delim.push_back(symbol);
+EXTERNC int keyHexToDecimal(char *skey_hex, char *skey_dec_out) {
+  mpz_t skey;
+  mpz_init(skey);
+  try {
 
-    size_t prev = 0, pos = 0;
-    do {
-      pos = str.find(delim, prev);
-      if (pos == std::string::npos)
-        pos = str.length();
-      std::string token = str.substr(prev, pos - prev);
-      if (!token.empty()) {
-        libff::alt_bn128_Fq coeff(token.c_str());
-        result.push_back(coeff);
-      }
-      prev = pos + delim.length();
-    } while (pos < str.length() && prev < str.length());
+    if (mpz_set_str(skey, skey_hex, 16) == -1) {
+      mpz_clear(skey);
+      LOG_ERROR("Could not convert hexadecimal into number");
+      return 1;
+    }
```

Comment:
Why is 1 returned here? In general the function should either throw an exception or return an error code as a separate variable